staging/src/k8s.io/apiserver/pkg/authentication/request/x509/verify_options.go

```diff
@@ -47,3 +47,25 @@ func NewStaticVerifierFromFile(clientCA string) (VerifyOptionFunc, error) {
 
 	return StaticVerifierFn(opts), nil
 }
+
+// DynamicStringSlice is a way to get a string slice value.  It is heavily used for authentication headers among other places.
+type DynamicStringSlice interface {
```

Comment:
It's weird that StaticStringSlice implements DynamicStringSlice. Maybe StringSliceProvider would be a better name. Why not just use a `func() []string` and skip the interface? Is this ever going to have any more methods?

Reply:
My understanding of the above comment was `s/DynamicStringSlice/StringSliceProvider` instead of `s/StaticStringSlice/StringSliceProvider`, `StaticStringSlice` describes the nature of the value better IMO.

Reply:
ah, that makes more sense.  Done